stdlib/--check-bounds=no/Project.toml

```diff
@@ -0,0 +1,2 @@
+name = "--check-bounds=no"
```

Comment:
The name is almost to cute. Maybe simply: `CheckBoundsNo`?

One other thought I just had is that it is kinda weird to use "distinct" packages for this kind of options settings.
Could we instead use Preferences? That already interacts with the caching subsystem.

cc: @staticfloat

Reply:
There needs to be a cache file that stores the curing definitions. Preferences don't (currently) have any mechanism that has identity for a cache file.

Reply:
So I was thinking that we would have one stdlib instead of two.

```
module CheckBounds

const my_uuid = #
Base.record_compiletime_preference(my_uuid, "check-bounds")
let d = Base.get_preferences(my_uuid)
    global const check_bounds = d["check-bounds"]
end

@Base.recompile_invalidations begin
    Core.should_check_bounds(boundscheck::Bool) = check_bounds
end
```

Compiletime preferences split the cache and you get the same curing effect.

Reply:
Elliot and I were discussing that we might want to add Preferences usage for Julia instead of pushing more and more stuff into command line flags. 

So while the user could use `julia --check-bounds=no`, they could also use their local LocalPreferences.toml to set

```
[CheckBounds]
check-bounds=no
```

I kinda like that from a user interface perspective.

Can you explain why the `precompile(:mandatory)` a bit more? I am not sure what problem it solves.

Reply:
So what #50578 turned into an explicit error. Gotcha